### ISSUE TITLE
[Bug]: aria/ selector returns elements not in the accessibility snapshot

### ISSUE THREAD

**Issue by braddunbar** [puppeteer/puppeteer#7594]
### Bug description

Steps to reproduce the problem:

1. Run the script below.
2. Expect the `aria-hidden` button to be ignored by the `aria/` selector.
3. The `aria/` selector actually returns the `aria-hidden` element.

```
import puppeteer from 'puppeteer'

const browser = await puppeteer.launch({
  args: ['--no-sandbox'],
  defaultViewport: {
    width: 1920,
    height: 1080,
  },
})

const page = await browser.newPage()

await page.setContent(`
  <button aria-hidden="true">Test</button>
  <button>Test</button>
`)

console.log(await page.$$eval('aria/Test', elements => (
  elements.map(element => element.outerHTML)
)))
console.log(await page.accessibility.snapshot())

await browser.close()
```

### Puppeteer version

10.0.0

### Node.js version

16.8.0

### npm version

7.21.0

### What operating system are you seeing the problem on?

macOS

### Relevant log output

```shell
$ node bug.js
[
  '<button aria-hidden="true" tabindex="-1">Test</button>',
  '<button>Test</button>'
]
{
  role: 'RootWebArea',
  name: '',
  children: [ { role: 'button', name: 'Test' } ]
}
```

**Comment by braddunbar** [comment]
*Note: I realize buttons are usually focusable elements and thus [shouldn't](https://www.w3.org/TR/using-aria/#fourth) get `aria-hidden="true"`, but this button has `tabindex="-1"` and is not focusable.*

**Comment by jschfflr** [comment]
@johanbay Could you take a look at this and check if this is working as intended or could be improved upon?

**Comment by johanbay** [comment]
This is actually intended. We decided to let the handler also query nodes marked [ignored](https://github.com/puppeteer/puppeteer/blob/f2e19276acb80e596ff5c781c9ae2bc2f2a8f363/src/common/Accessibility.ts#L247) in the tree to make the handler more versatile.
@braddunbar can you expand a bit more on your use case? Are you querying the button to ensure that it truly was hidden?

**Comment by braddunbar** [comment]
Hey @johanbay - thanks for the response!

We're doing accessibility-aware testing with puppeteer, using snapshots and `aria/` selectors to find elements by the text that's available via screen reader. In this particular case, we were looking for a button by text and getting a collision with a separate button with `aria-hidden="true"`. The second button exists mainly for reasons of simplicity and styling.

We've disambiguated in the meantime by filtering out elements with `aria-hidden="true"`, but we were definitely surprised by the behavior. That said, if it's intentional we can continue with our workaround.

**Comment by jschfflr** [comment]
@johanbay I would also expect the API to behave the same in both scenarios. Is there maybe a way to add a flag that can toggle the inclusion of ignored nodes?

**Comment by johanbay** [comment]
I think it would be included in the snapshot if [interestingOnly](https://github.com/puppeteer/puppeteer/blob/f863f4bfe015e57ea1f9fbb322f1cedee468b857/src/common/Accessibility.ts#L102) is set to `false`.
We could include this as a querying attribute for `aria/`? So one would do `aria/Test[ignored:false]` in this case.

**Comment by andyearnshaw** [comment]
I think the default should be at least consistent across both API methods. Rather than `[ignored:false]` or `[ignored="false"]` which don't really match up with other ARIA properties that might be included in `[]` delimiters in the future, it might be more pertinent to make _interesting nodes only_ the default and use a different prefix where _uninteresting_ nodes are required, e.g. `aria-all/`. This would be a breaking change, though.

**Comment by jschfflr** [comment]
@andyearnshaw I'd agree on getting the default consistent.
@johanbay What do you think about the proposed change?

**Comment by andyearnshaw** [comment]
This is still valid. Also, I'm happy to help out on it if there's a solid plan on the implementation.

**Comment by gorshkov-leonid** [comment]
It is still actual. My case external lib with license label. Webcomponent is closed and I cannot remove or overlap this plate.
When I use **Capture node screenshot** via dev tool I get what is expected - only content without this license label. 
But puppteer returns what is shown. Everything would be fine if the lable was displayed deterministically
